src/mbgl/map/tile_worker.cpp

```diff
@@ -142,14 +157,17 @@ void TileWorker::parseLayer(const StyleLayer* layer, const GeometryTile& geometr
                                      spriteStore,
                                      glyphAtlas,
                                      glyphStore,
-                                     *collisionTile,
                                      mode);
 
     std::unique_ptr<Bucket> bucket = layer->createBucket(parameters);
 
-    if (layer->is<SymbolLayer>() && partialParse) {
-        // We cannot parse this bucket yet. Instead, we're saving it for later.
-        pending.emplace_back(layer->as<SymbolLayer>(), std::move(bucket));
+    if (layer->is<SymbolLayer>()) {
+        if (partialParse) {
+            // We cannot parse this bucket yet. Instead, we're saving it for later.
+            pending.emplace_back(layer->as<SymbolLayer>(), std::move(bucket));
+        } else {
+            placementPending.emplace(layer->bucketName(), std::move(bucket));
```

Comment:
Since label names are unique, there should never be anything to overwrite. Also, this is the same as [this](https://github.com/mapbox/mapbox-gl-native/blob/4515bb00c0fd6c90050cf3befc4e848aaa7362c0/src/mbgl/map/tile_worker.cpp#L1600)